### PR TITLE
use `ExternalMemoryAdjuster` on zlib allocations

### PR DESCRIPTION
**Pull request by anonrig**
Let's also inform v8 about the allocations that we've made.

### CODE REVIEW

File: src/workerd/api/node/zlib-util.h [hunk 1]
```diff
@@ -291,17 +291,23 @@ class ZlibUtil final: public jsg::Object {
...
+  // isolate pointer and use that to get the external memory adjustment.
   class Allocator final {
   public:
     static void* AllocForZlib(void* data, uInt items, uInt size);
     static void* AllocForBrotli(void* data, size_t size);
     static void FreeForZlib(void* data, void* pointer);
 
+    struct Allocation {
+      kj::Array<kj::byte> data;
+      kj::Maybe<jsg::ExternalMemoryAdjustment> memory;
```
Comment (npaun) [inline src/workerd/api/node/zlib-util.h:306]:
```suggestion
      kj::Maybe<jsg::ExternalMemoryAdjustment> memoryAdjustment;
```

nit: Makes it clearer which field is the allocation and what is the metadata.

**Review by npaun — APPROVED** [review]
LGTM but definitely needs a review from @jasnell too

**Review by jasnell — APPROVED** [review]

**Comment by jasnell** [comment]
Let's make sure to run an internal CI run on this to confirm. Should be just fine but let's make sure.

**Comment by anonrig** [comment]
I've reimplemented the allocator and moved it to api/streams, since the compression there also needs to be tracked using v8. appreciate a re-review @jasnell @npaun

File: src/workerd/api/node/BUILD.bazel [hunk 2]
```diff
@@ -14,7 +14,6 @@ wd_cc_library(
         "@capnp-cpp//src/kj/compat:kj-gzip",
         "@nbytes",
         "@simdutf",
-        "@zlib",
```
Comment (npaun) [inline src/workerd/api/node/BUILD.bazel:17]: What's the reason for removing this? Does kj-gzip cover it?

File: src/workerd/api/streams/compression.c++ [hunk 3]
```diff
@@ -3,14 +3,51 @@
...
+#include "nbytes.h"
 #include <workerd/io/features.h>
-#include <zlib.h>
 #include <deque>
 #include <vector>
 #include <iterator>
 
 namespace workerd::api {
 
+void CompressionAllocator::configure(z_stream* stream) {
```
Comment (npaun) [inline src/workerd/api/streams/compression.c++:15]: Maybe call it configureForZlib, as that's the only use?

File: src/workerd/api/node/BUILD.bazel (hunk 2 above)
Comment (anonrig) [reply src/workerd/api/node/BUILD.bazel:17]: Yes, this also adds lots of warnings to edgeworker.

**Review by npaun — APPROVED** [review]
Re-approved

**Comment by anonrig** [comment]
Had to revert the v8 tracking changes due to an internal bug. I'll investigate differently. Meanwhile, the PR includes a little bit of refactoring to simplify the function signatures.

**Pull request merged**